Fix failing unit_tests/ensemble_evaluator/test_ensemble_legacy.py::test_run_and_cancel_legacy_ensemble

**Describe the bug**
The following test fails, but only when running it as github workflow:
```python
@pytest.mark.timeout(60)
@pytest.mark.scheduler
    def test_run_and_cancel_legacy_ensemble(
        tmpdir, make_ensemble_builder, monkeypatch, try_queue_and_scheduler
    ):
        num_reals = 2
        custom_port_range = range(1024, 65535)
        with tmpdir.as_cwd():
            ensemble = make_ensemble_builder(
                monkeypatch, tmpdir, num_reals, 2, job_sleep=40
            ).build()
            config = EvaluatorServerConfig(
                custom_port_range=custom_port_range,
                custom_host="127.0.0.1",
                use_token=False,
                generate_cert=False,
            )
    
            evaluator = EnsembleEvaluator(ensemble, config, 0)
    
            evaluator._start_running()
            with Monitor(config) as mon:
                cancel = True
                with contextlib.suppress(
                    ConnectionClosed
                ):  # monitor throws some variant of CC if dispatcher dies
>                   for _ in mon.track():

unit_tests/ensemble_evaluator/test_ensemble_legacy.py:78: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
/prog/res/komodo/bleeding-py38-rhel7/root/lib64/python3.8/site-packages/ert/ensemble_evaluator/monitor.py:95: in track
    for message in duplexer.receive():
/prog/res/komodo/bleeding-py38-rhel7/root/lib64/python3.8/site-packages/ert/ensemble_evaluator/sync_ws_duplexer.py:126: in receive
    event = asyncio.run_coroutine_threadsafe(
/opt/rh/rh-python38/root/usr/lib64/python3.8/concurrent/futures/_base.py:439: in result
    self._condition.wait(timeout)
self = <Condition(<unlocked _thread.RLock object owner=0 count=0 at 0x7f1aad32b450>, 0)>
timeout = None

    def wait(self, timeout=None):
        """Wait until notified or until a timeout occurs.
    
        If the calling thread has not acquired the lock when this method is
        called, a RuntimeError is raised.
    
        This method releases the underlying lock, and then blocks until it is
        awakened by a notify() or notify_all() call for the same condition
        variable in another thread, or until the optional timeout occurs. Once
        awakened or timed out, it re-acquires the lock and returns.
    
        When the timeout argument is present and not None, it should be a
        floating point number specifying a timeout for the operation in seconds
        (or fractions thereof).
    
        When the underlying lock is an RLock, it is not released using its
        release() method, since this may not actually unlock the lock when it
        was acquired multiple times recursively. Instead, an internal interface
        of the RLock class is used, which really unlocks it even when it has
        been recursively acquired several times. Another internal interface is
        then used to restore the recursion level when the lock is reacquired.
    
        """
        if not self._is_owned():
            raise RuntimeError("cannot wait on un-acquired lock")
        waiter = _allocate_lock()
        waiter.acquire()
        self._waiters.append(waiter)
        saved_state = self._release_save()
        gotit = False
        try:    # restore state no matter what (e.g., KeyboardInterrupt)
            if timeout is None:
>               waiter.acquire()
E               Failed: Timeout >60.0s

/opt/rh/rh-python38/root/usr/lib64/python3.8/threading.py:302: Failed
```

**To reproduce**
It fails on gh action workflows both on PR runs and komodo tests. See examples:
ert-PR: https://github.com/equinor/ert/actions/runs/7208911337/job/19638817158
komodo: https://github.com/equinor/komodo-releases/actions/runs/7391951454/job/20110678281

**Expected behaviour**
The test should pass.

**Screenshots**
If applicable, add screenshots to help explain your problem.

**Environment**
 - OS: [e.g. RHEL7]
 - ERT/Komodo release: bleeding (constantly), PR (flaky behaviour)
 - Python version
 - Remote/HPC execution involved: [yes|no]

--- a/src/ert/scheduler/local_driver.py
+++ b/src/ert/scheduler/local_driver.py
@@ -26,7 +26,7 @@ class LocalDriver(Driver):
             self._tasks[iens].cancel()
             await self._tasks[iens]
             del self._tasks[iens]
-        except KeyError:
+        except (KeyError, asyncio.CancelledError):
             return
 
     async def finish(self) -> None:

